src/Expander.php

```diff
@@ -241,6 +241,14 @@ public static function expandPropertyWithReferenceData(
      */
     public static function expandProperty($property_name, $unexpanded_value, $data)
     {
+        if (strpos($property_name, "env.") === 0 &&
+          !$data->has($property_name)) {
+            $env_key = substr($property_name, 4);
+            if (!empty($_ENV) && array_key_exists($env_key, $_ENV)) {
```

Comment:
Not sure why, but getenv() fails here.

Reply:
By fails, do you mean does not work with your test where you set `$_ENV['test']`? Try with getenv here and putenv in the test. getenv is far superior to $_ENV, as the former should in theory always work, whereas $_ENV is not always populated.

Reply:
> By fails, do you mean does not work with your test where you set $_ENV['test']? Try with getenv here and putenv in the test.

But isn't it problematic that users can't set env vars without putenv()?

Reply:
This is exactly as you want it to be. Typically, environment variables will be provided from outside the application, not inside it. As previously mentioned, `$_ENV` is not a reliable way to read environment variables; if you want to set an environment variable from inside a program, you should use putenv. Setting `$_ENV` is relying on a side-effect, as `$_ENV` is just a shadow of the environment system.

Reply:
(In other words, I am strongly advocating for `getenv`)

Reply:
I see. So, I've really been using PHP env vars incorrectly!

I found [this comment](http://php.net/manual/en/function.putenv.php#98540) enlightening:

> putenv/getenv, $_ENV, and phpinfo(INFO_ENVIRONMENT) are three completely distinct environment stores. doing putenv("x=y") does not affect $_ENV; but also doing $_ENV["x"]="y" likewise does not affect getenv("x"). And neither affect what is returned in phpinfo().